### PR TITLE
Set mimeTypeDetector nullable by default

### LINKED ISSUE #266: Google Cloud Storage dependency injection error after updating the adapter
### Bug Report

This [Google Cloud Storage adapter commit](https://github.com/thephpleague/flysystem-google-cloud-storage/commit/b698f407391f63149fe6ba4b950d9fd667aa4da5) introduced a new constructor parameter (`$mimeTypeDetector`) leading to a dependency injection error.

|    Q        |   A
|------------ | ------
| BC Break    | yes
| Version     | 4.4.1

#### Summary

Update `league/flysystem-google-cloud-storage` from `3.0.9` to `3.0.15` with  `oneup/flysystem-bundle` `4.4.1`

Resulting error :

```
League\Flysystem\GoogleCloudStorage\GoogleCloudStorageAdapter::__construct(): Argument #5 ($mimeTypeDetector) must be of type ?League\MimeTypeDetection\MimeTypeDetector, string given, called in /app/var/cache/dev/ContainerAIbli  
  xC/srcApp_KernelDevDebugContainer.php on line 5387 
```

#### How to reproduce

Without touching configuration, a working project does not work after update.

I managed to fix with this workaround in `Kernel.php` : 

```
public function process(ContainerBuilder $container)
{
    $container->getDefinition('oneup_flysystem.adapter.googlecloudstorage')->setArgument(4, null);
}
```
This issue has also been posted in [the flysystem repo](https://github.com/thephpleague/flysystem/issues/1447)

### PR DESCRIPTION
**Pull request by jeremyhalin**
Fixes #266 
Fixes https://github.com/thephpleague/flysystem/issues/1447

I'm not sure about what I'm doing but I'll try to fix this BC.

### CODE REVIEW

File: src/DependencyInjection/Factory/Adapter/GoogleCloudStorageFactory.php [hunk 1]
```diff
@@ -31,6 +31,7 @@ public function create(ContainerBuilder $container, string $id, array $config):
             ->replaceArgument(1, $config['prefix'])
             ->replaceArgument(2, $config['visibilityHandler'])
             ->replaceArgument(3, $config['defaultVisiblity'])
+            ->replaceArgument(4, null)
```
Comment (bytehead) [inline src/DependencyInjection/Factory/Adapter/GoogleCloudStorageFactory.php:34]: Is there a reason why the `mimeTypeDetector` should not be configurable?

Comment (jeremyhalin) [reply src/DependencyInjection/Factory/Adapter/GoogleCloudStorageFactory.php]: I asked myself the question. I guess no.

**Comment by jeremyhalin** [comment]
Thanks for reviewing. When do you think we can merge and release this fix?

**Comment by bytehead** [comment]
Will have a look tomorrow, ok?

**Review by bytehead — APPROVED** [review]

**Comment by bytehead** [comment]
Thank you very much @jeremyhalin ❤️

**Comment by bytehead** [comment]
[`4.4.2`](https://github.com/1up-lab/OneupFlysystemBundle/releases/tag/4.4.2) is released 🥳

**Pull request merged**